.github/workflows/aznfs-test.yml

```diff
@@ -0,0 +1,354 @@
+name: Test AZNFS
+
+on:
+  workflow_dispatch:
+    inputs:
+      versionName:
+        description: 'Version Name'
+        required: true
+      storageAccounts:
+        description: 'Space-separated Storage Accounts'
+        required: true
+
+jobs:
+  run_shellcheck:
+    name: Shellcheck for AZNFS
+    runs-on: self-hosted-ubuntu22
+    steps:
+      - name: Install ShellCheck
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y shellcheck
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+      - name: Install AZNFS
+        shell: bash
+        run: |
+          export AZNFS_NONINTERACTIVE_INSTALL=1
+          sudo wget -O - -q "https://github.com/Azure/AZNFS-mount/releases/download/${{ github.event.inputs.versionName }}/aznfs_install.sh" | bash
+      - name: Run Shellcheck on Scripts
+        shell: bash
+        run: |
+          declare -a files_to_check=(
+            "/src/aznfswatchdog"
+            "/src/mountscript.sh"
+            "/lib/common.sh"
+            "/scripts/aznfs_install.sh"
+          )
+          for file in "${files_to_check[@]}"; do
+            echo "=== Checking $file for errors ==="
+
+            full_file_path="$GITHUB_WORKSPACE$file"
+            if output=$(shellcheck --severity=error "$full_file_path"); then
+              echo "No errors found in $file."
+            else
+              echo "$output"
+            fi
+          done
+
+  test_aznfs_ubuntu18:
+    name: Test AZNFS - Ubuntu 18
+    runs-on: self-hosted-ubuntu18
+    needs: run_shellcheck
+    steps:
+      - name: Install Prerequisites
+        shell: bash
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y build-essential
+      - name: Check Version and Storage Account
+        shell: bash
+        run: |
+          versionName=${{ github.event.inputs.versionName }}
+          storageAccounts="${{ github.event.inputs.storageAccounts }}"
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+      - name: Untar UnixTestSuite
+        shell: bash
+        run: |
+          mkdir -p /lib
+          tar -xf $GITHUB_WORKSPACE/UnixTestSuite.tar -C /lib
+      - name: Run Test Script
+        shell: bash
+        run: |
+          export RELEASE_NUMBER=${{ github.event.inputs.versionName }}
+          export STORAGE_ACCOUNTS="${{ github.event.inputs.storageAccounts }}"
+          chmod +x $GITHUB_WORKSPACE/testing/testscript.sh
+          $GITHUB_WORKSPACE/testing/testscript.sh
+
+  test_aznfs_ubuntu20:
+    name: Test AZNFS - Ubuntu 20
+    runs-on: self-hosted-ubuntu20
+    needs: test_aznfs_ubuntu18
+    steps:
+      - name: Install Prerequisites
+        shell: bash
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y build-essential
+      - name: Check Version and Storage Account
```

Comment:
can we create a composite action (https://docs.github.com/en/actions/creating-actions/creating-a-composite-action) called prep-aznfs-test-machine which runs a shell script that correctly preps the machine on which it runs. 
Then we just call that action from here followed by the actual test call.
This will reduce code repetition.